R/s3.put.R

```diff
@@ -17,10 +19,17 @@ s3.put <- function (x, path, bucket.location = "US", verbose = FALSE,
   on.exit(unlink(x.serialized, force = TRUE), add = TRUE)
   saveRDS(x, x.serialized)
 
-  s3.cmd <- paste("put", x.serialized, paste0('"', path, '"'), ifelse(encrypt,
+  s3.cmd <- paste("put", x.serialized, paste0('"', s3key, '"'), ifelse(encrypt,
       "--encrypt", ""), paste("--bucket-location", bucket.location),
       ifelse(verbose, "--verbose --progress", "--no-progress"), ifelse(debug,
           "--debug", ""), '--check-md5')
 
   system2(s3cmd(), s3.cmd, stdout = TRUE)
+  if (check_exists) {
+    retry_count <- 0
+    while (!s3exists(name, path) && retry_count < num_retries) {
+        system2(s3cmd(), s3.cmd, stdout = TRUE)
+        retry_count <- retry_count + 1
+    }
+  }
```

Comment:
```r
if (isTRUE(check_exists) && num_retries > 0) {
  Recall(x, path, bucket.location, verbose, debug, encrypt, check_exists, num_retries - 1)
}
```

and maybe figure out how to abuse match.call in order to call Recall without further verbosity